### PR TITLE
restructure migraphx

### PR DESCRIPTION
**Pull request by SwRaw**

### CODE REVIEW

**Comment by pfultz2** [comment]
This is a copy of the dev introduciton. We do NOT want to use that to introduce migraphx since the APIs are different from the user APIs. We should be pointing users to the user API not the internal API.

**Review by randyh62 — APPROVED** [review]
I think you need to add the License back to the _toc.yml.in, but otherwise it looks good to me.

**Review by causten — COMMENTED** [review]
The license checker is failing because ['docs/conf.py'] is not set to 2024

**Comment by SwRaw** [comment]
> The license checker is failing because ['docs/conf.py'] is not set to 2024

@causten Could you please let me know what changes need to be made to conf.py I checked and I could not find any reference to another year.
We have :
copyright = "Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved."

**Comment by SwRaw** [comment]
> I think you need to add the License back to the _toc.yml.in, but otherwise it looks good to me.
@randyh62 
I removed it because we already have the following statement on the index page:
Licensing information for all ROCm components is listed on our `Licensing <https://rocm.docs.amd.com/en/latest/about/license.html>`_ page.
The previously used license points to another file. We don't want two different references to license.

**Comment by causten** [comment]
> > The license checker is failing because ['docs/conf.py'] is not set to 2024
> 
> @causten Could you please let me know what changes need to be made to conf.py I checked and I could not find any reference to another year. We have : copyright = "Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved."

No you don't, not in this PR.  If you look at docs/conf.py in this PR and look at the top of the file you will see 2023

File: docs/driver-options.rst [hunk 1]
```diff
@@ -1,7 +1,13 @@
-MIGraphX Driver
+.. meta::
+   :description: MIGraphX provides an optimized execution engine for deep learning neural networks
+   :keywords: MIGraphX, ROCm, library, API, tool
+
+.. _driver-options:
+
+Driver options
```
Comment (pfultz2) [inline docs/driver-options.rst:7]: This file should go under the reference directory.

File: docs/migraphx-driver.rst [hunk 2]
```diff
@@ -0,0 +1,650 @@
...
+   :description: MIGraphX provides an optimized execution engine for deep learning neural networks
+   :keywords: MIGraphX, ROCm, library, API, tool
+
+.. _migraphx-driver:
+
+=====================
+MIGraphX driver
+=====================
+
+The MIGraphX driver is a command-line tool that allows you to utilize many of the MIGraphX core functions without having to write a program.
```
Comment (pfultz2) [inline docs/migraphx-driver.rst:11]: This file should go under the how-to guide or turtorials section.

File: docs/contributing-to-migraphx.rst [hunk 3]
```diff
@@ -0,0 +1,161 @@
+.. meta::
+   :description: MIGraphX provides an optimized execution engine for deep learning neural networks
+   :keywords: MIGraphX, ROCm, library, API
+
+.. _contributing-to-migraphx:
+
+==========================
+Contributing to MIGraphX
```
Comment (pfultz2) [inline docs/contributing-to-migraphx.rst:8]: This file should go under the dev directory.

**Comment by pfultz2** [comment]
Is there a way to see a preview of the documentation?

**Comment by causten** [comment]
https://advanced-micro-devices-demo--2711.com.readthedocs.build/projects/AMDMIGraphX/en/2711/dev/data.html
-- there is a warning that needs to be fixed.

**Comment by SwRaw** [comment]
> https://advanced-micro-devices-demo--2711.com.readthedocs.build/projects/AMDMIGraphX/en/2711/dev/data.html -- there is a warning that needs to be fixed.

That is not really a warning but a message that the page is generated for a particular pull request and it is not the final/live page

**Comment by causten** [comment]
> > https://advanced-micro-devices-demo--2711.com.readthedocs.build/projects/AMDMIGraphX/en/2711/dev/data.html -- there is a warning that needs to be fixed.
> 
> That is not really a warning but a message that the page is generated for a particular pull request and it is not the final/live page

Scroll down and you will see what I am talking about.  Here is a better link that gets you closer to the message...
https://advanced-micro-devices-demo--2711.com.readthedocs.build/projects/AMDMIGraphX/en/2711/dev/data.html#tensor-view

**Comment by SwRaw** [comment]
> > > https://advanced-micro-devices-demo--2711.com.readthedocs.build/projects/AMDMIGraphX/en/2711/dev/data.html -- there is a warning that needs to be fixed.
> > 
> > 
> > That is not really a warning but a message that the page is generated for a particular pull request and it is not the final/live page
> 
> Scroll down and you will see what I am talking about. Here is a better link that gets you closer to the message... https://advanced-micro-devices-demo--2711.com.readthedocs.build/projects/AMDMIGraphX/en/2711/dev/data.html#tensor-view

Yeah, saw that. It is related to Doxygen not to my changes. Perhaps, somebody from the team needs to look into this.

**Review by causten — APPROVED** [review]

**Pull request merged**